src/ai/backend/manager/registry.py

```diff
@@ -2144,6 +2144,10 @@ async def destroy_session(
                     await self.event_producer.produce_event(
                         SessionTerminatingEvent(session_id, reason),
                     )
+                case SessionStatus.TERMINATED:
+                    raise GenericForbidden(
+                        "Cannot destroy sessions that has already been already terminated"
+                    )
                 case _:
                     await SessionRow.set_session_status(
```

Comment:
Just a question: is it okay to leave `CANCELLED` as is?

Reply:
good point! I updated this part to restrict destroy cancelled sessions also